.ci/Jenkinsfile

```
@@ -184,19 +184,26 @@ pipeline {
               unstash 'source'
               dir("${BASE_DIR}") {
                 script {
-                  def suiteParam = params.runTestsSuite
-                  def suites = readYaml(file: '.ci/.e2e-tests.yaml')
+                  def suitesParam = params.runTestsSuites
+                  def existingSuites = readYaml(file: '.ci/.e2e-tests.yaml')
                   def parallelTasks = [:]
-                  suites['SUITES'].each { item ->
-                    def suite = item.suite
-                    def feature = item.feature
-                    if (suiteParam == "all" || suiteParam == suite) {
-                      def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go" ]
-                      if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
-                        log(level: 'INFO', text: "Adding ${suite}:${feature} test suite to the build execution")
-                        parallelTasks["${suite}_${feature}"] = generateFunctionalTestStep(suite: "${suite}", feature: "${feature}")
-                      } else {
-                        log(level: 'WARN', text: "The ${suite}:${feature} test suite won't be executed because there are no modified files")
+
+                  if (suitesParam == "") {
+                    log(level: 'DEBUG', text: "Iterate through existing test suites")
+                    existingSuites['SUITES'].each { item ->
+                      def existingSuite = item.suite
+                      def tags = item.tags
+                      checkTestSuite(parallelTasks, existingSuite, tags)
+                    }
+                  } else {
+                    log(level: 'DEBUG', text: "Iterate through the comma-separated test suites (${suitesParam}), comparing with the existing test suites")
+                    suitesParam.split(',').each { suiteParam ->
+                      existingSuites['SUITES'].each { item ->
+                        def existingSuite = item.suite
+                        if (suiteParam.trim() == existingSuite) {
+                          def tags = item.tags
+                          checkTestSuite(parallelTasks, existingSuite, tags)
+                        }
```

You can simplify the collection with findAll

```suggestion
                      existingSuites['SUITES'].findAll { suiteParam.trim() == it.suite }.each { item ->
                          checkTestSuite(parallelTasks, existingSuite, item.tags)
```

---

Like it!